acceptancetests/jujupy/workloads.py

```diff
@@ -0,0 +1,141 @@
+# This file is part of JujuPy, a library for driving the Juju CLI.
+# Copyright 2013-2018 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the Lesser GNU General Public License version 3, as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+# SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the Lesser
+# GNU General Public License for more details.
+#
+# You should have received a copy of the Lesser GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import print_function
+
+from urllib2 import urlopen
+from jujucharm import local_charm_path
+import logging
+import os
+import requests
+
+from jujupy.wait_condition import (
+    AgentsIdle,
+    AllApplicationActive,
+    AllApplicationWorkloads,
+    )
+from jujupy.utility import (
+    get_unit_ipaddress,
+    temp_dir,
+)
+
+
+__metaclass__ = type
+
+log = logging.getLogger(__name__)
+
+
+def deploy_mediawiki_with_db(client):
+    client.deploy('cs:mysql')
+    client.wait_for_started()
+    client.wait_for_workloads()
+
+    client.deploy('cs:mediawiki')
+    client.wait_for_started()
+    # mediawiki workload is blocked ('Database needed') until a db
+    # relation is successfully made.
+    client.juju('relate', ('mediawiki:db', 'mysql:db'))
+    client.wait_for_workloads()
+    client.wait_for_started()
+    client.juju('expose', 'mediawiki')
+    client.wait_for_workloads()
+    client.wait_for_started()
+
+
+def assert_media_wiki_is_responding(client):
+    log.debug('Assert media wiki is responding.')
+    status = client.get_status()
+    [wiki_unit_name] = [
+        k for k, v in status.get_applications()['mediawiki']['units'].items()
+        if v.get('leader', False)]
+    wiki_ip = get_unit_ipaddress(client, wiki_unit_name)
+    resp = requests.get('http://{}'.format(wiki_ip))
+    if not resp.ok:
+        raise AssertionError('Mediawiki not responding; {}: {}'.format(
+            resp.status_code, resp.reason
+        ))
+    if '<title>Please set name of wiki</title>' not in resp.content:
```

Comment:
If possible let's make this resistant to minor arbitrary string changes. Could we check the response header instead? Could we consider a /ok and no error code to be sufficient?